addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
So, how I think this ends up being played out is:

I want to wipe out the cache based on a predicate, not a delay.

Nothing is stopping from a component to

```
willDestroy() {
  setTimeout(() => {
    this.cache.clearCache();
  }, 50)
}
```

Basically, I'm wondering if this logic should go here